crates/ruff_python_parser/src/lexer.rs

```diff
@@ -809,19 +804,22 @@ impl<'src> Lexer<'src> {
         TokenKind::String
     }
 
-    pub fn next_token(&mut self) -> Token {
-        self.current = self.next_token_impl();
-        self.current
+    /// Lex the next token.
+    pub(crate) fn next_token(&mut self) -> TokenKind {
+        self.cursor.start_token();
+        self.current_kind = self.lex_token();
```

Comment:
I think we should reset `current_value` to `TokenValue::None` to prevent that a value keeps lingering if it isn't consumed using `take_value`
```suggestion
				self.current_value = TokenValue::None;
        self.current_kind = self.lex_token();
```